apps/api/src/app/shared/framework/idempotency.interceptor.ts

```diff
@@ -0,0 +1,122 @@
+// idempotency.interceptor.ts
+
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+  UnprocessableEntityException,
+  HttpException,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { CacheService } from '@novu/application-generic';
+import { Observable, of, throwError } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+const LOG_CONTEXT = 'IdempotencyInterceptor';
+const IDEMPOTENCY_CACHE_TTL = 60 * 60 * 24; //24h
```

Comment:
Lets move this down to 6 hours, as we get 1 quarter of a million or more request per hour and this could put alot of strain on our redis service.